handlers/rootfs_image.go

```diff
@@ -139,7 +140,26 @@ func (rp *Rootfs) ReadHeader(r io.Reader, path string, version int, augmented bo
 		}
 
 	case filepath.Base(path) == "meta-data":
-		// TODO: implement when needed
+		dec := json.NewDecoder(r)
+		var data interface{}
+		err := dec.Decode(&data)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return errors.Wrap(err, "error reading meta-data")
+		}
+		jsonObj, ok := data.(map[string]interface{})
```

Comment:
Yes, for this JSON for example:

```
[
  "a",
  "b"
]
```

In this case it will be a `[]string`. This is explicitly disallowed because of the [augmented artifacts feature](https://github.com/mendersoftware/mender/blob/master/Documentation/update-modules-v3-file-api.md#filtering-header-fields), which we have not implemented yet, but which we probably will. When using augmented artifacts, the top object must be a JSON object (`map[string]interface{}`) for field filtering to work correctly.

Reply:
Smrt!